core/templates/pages/topic-viewer-page/topic-viewer-page.component.ts

```diff
@@ -127,6 +129,14 @@ export class TopicViewerPageComponent implements OnInit {
 
   setActiveTab(newActiveTabName: string): void {
     this.activeTab = newActiveTabName;
+    if (newActiveTabName === 'story') {
+      console.error('me me');
+      this.windowRef.nativeWindow.location.hash = 'lessons';
+    } else if (newActiveTabName === 'practice') {
+      this.windowRef.nativeWindow.location.hash = 'practice';
+    } else {
+      this.windowRef.nativeWindow.location.hash = 'revision';
+    }
```

Comment:
I couldn't find  screenshot / video showing the working of the URL changes. Could you please attach it? 

I think there might be an error in setting the URL here. Per the current logic, we navigate to the corresponding tab based on the last URL subpart (_not_ the hash value). Example:

`/learn/math/place-values/story` -> selects story tab
`/learn/math/place-values/practice` -> selects practice tab
`/learn/math/place-values/revision` -> selects revision tab

the issue with the current thing is, if you go to `/story` and select the practice tab, the URL should change to `/practice` (ditto for others) but it doesn't. I suspect with the changes in this PR, we will be changing the location hash and not the actual URL (i.e. selecting practice tab will become `/story#practice` instead of `/practice`). Could you please check this? Thanks!

Reply:
Thanks @kevintab95 . I have initially only checked for `/learn/math/place-values#practice` , `/learn/math/place-values#lessons` , `/learn/math/place-values#revision` . Now I have changed the current logic to set active tab from url using the hash value and no longer the url subpart to maintain uniformity i.e `/learn/math/place-values#lessons` -> selects story tab and similar for practice and revision.